splash/tests/test_execute.py

```diff
@@ -3456,3 +3453,145 @@ def test_enable_and_disable_private_mode(self):
         data = resp.json()
         self.assertIn(u'world of splash', data["html1"])
         self.assertNotIn(u"world of splash", data["html2"])
+
+
+class MouseEventsTest(BaseLuaRenderTest):
+
+    def _assert_event_property(self, name, value, resp):
+        self.assertIn("{}:{}".format(name, value), resp.text)
+
+    def test_click(self):
+        resp = self.request_lua("""
+             function main(splash)
+                assert(splash:go(splash.args.url))
+                get_dimensions = splash:jsfunc([[
+                    function () {
+                        rect = document.getElementById('button').getBoundingClientRect();
+                        return {"x":rect.left, "y": rect.top}
+                    }
+                ]])
+                dimensions = get_dimensions()
+                assert(splash:go(splash.args.url))
+                splash:mouse_click(dimensions.x, dimensions.y)
+                splash:wait(0.1)
+                return splash:html()
+            end
+            """, {"url": self.mockurl("jsevent?event_type=click")})
+        self.assertStatusCode(resp, 200)
+        self.assertIn("button", resp.text)
+        self.assertNotIn('this must be removed after click', resp.text)
+        self._assert_event_property("type", "click", resp)
+
+    def test_click_with_bad_arguments(self):
+        resp = self.request_lua("""
+             function main(splash)
+                assert(splash:go(splash.args.url))
+                splash:mouse_click(nil, nil)
+                splash:wait(0.1)
+                return splash:html()
+            end
+            """, {"url": self.mockurl("jsevent?event_type=click")})
+        self.assertScriptError(resp, ScriptError.LUA_ERROR,
+                               'AssertionError()')
```

Comment:
Could you please make errors messages a bit more detailed? See e.g. how are errors raised in splash:go.